### ISSUE TITLE
Feature request: pnpm link -r

### ISSUE THREAD

**Issue by nategreen** [pnpm/pnpm#3026]
When running `pnpm link -r <dir>`, I expected `pnpm` to link the local package at `<dir>` (outside the workspace) to all the packages in my workspace. I found out that wasn't possible:

```bash
 ERROR  Unknown option: 'recursive'
```

It would be great if `pnpm link` (and `unlink`) could be recursive. A flag like the one suggested in #1228 would be good too—only link if the target package name is found in the workspace packages' dependencies.

**Comment by zkochan** [comment]
You can use a workaround. In the root `package.json` of your workspace, use `pnpm.overrides`. Let's say you want to link foo from a local directory, put this in the pnpm.overrides:

```js
{
  "pnpm": {
    "overrides": {
      "foo": "<dir>"
    }
}
```

then just run `pnpm install`.

**Comment by benallfree** [comment]
@zkochan This is not working as expected for me:

```
{
  "dependencies": {
     "foo": "0.0.1",
  },
  "pnpm": {
    "overrides": {
      "foo": "../foo"
    }
}
```

Still links to 

```
foo -> .pnpm/foo@0.0.1/node_modules/foo
```

**Comment by zkochan** [comment]
what about this:

```
{
  "dependencies": {
     "foo": "0.0.1",
  },
  "pnpm": {
    "overrides": {
      "foo": "link:../foo"
    }
}
```

**Comment by benallfree** [comment]
@zkochan Nope, same result :(

Further background... I ran into this because my I found relative paths in my `pnpm-lock.yaml` which was making CI fail. So I started looking for a way to support local package links without affecting the lockfile and that led me here. But actually, it would be best to have maybe a `.pnpm-local` or some kind of file that didn't get committed so I could maintain package links. If you think that's a worthwhile discussion I'll make a new ticket for it :)

**Comment by zkochan** [comment]
`pnpm.overrides` will result in relative paths in `pnpm-lock.yaml` as well. Why are modifications to `pnpm-lock.yaml` an issue? Because you accidentally committed them?

**Comment by benallfree** [comment]
@zkochan Well that's a good point, packages intended to be used as a dependency generally shouldn't have a lockfile committed. But top level packages like a monorepo may have a lockfile so a dev team can all work against the same deps and have repeatable builds. In those cases, CI would fail if the lockfile contained paths only present in a local dev environment. I assume this is why `npm link` doesn't modify the lockfile. 

I have a complex case where I'm trying to fork transient dependencies as well as direct dependencies. For example, my main monorepo depends upon a package I authored named `georedis-promised`. That package, in turn, depends upon a forked version of `node-georedis`. So my file structure looks like:

**Comment by zkochan** [comment]
> @zkochan Nope, same result. Is pnpm.overrides only active inside workspaces?

no, it works in non-workspaces as well. What pnpm version do you use? This feature was added in 5.10.1: https://pnpm.js.org/en/package_json#pnpmoverrides

**Comment by igortas** [comment]
@zkochan 
Is there other way to do this?
I've started using turborepo with pnpm and the structure is like this
- apps
- packages (here i have subfolders like shared, and in shared i've installed shared deps, like rimraf)

Problem is, in apps I see the symlink from shared folder, but runninig inside apps/*, command like:
"dev": "rimraf node_modules" does not working, command is not found.
Is there a way with pnpm to know upstreams in monorepo about subfolders commands , like from packages... or everything needs to be scopes in the root folder?

**Comment by zkochan** [comment]
@igortas this is not really related to this issue but if you need rimraf to be accessible in each package of the monorepo, then you need to install it as a dependencies in the root of the monorepo. 

Like in case of the repository, these packages are available in all subpackages: https://github.com/pnpm/pnpm/blob/6e4becd9977cb7d60c6614b5e2569394de9607aa/package.json#L27-L59

**Comment by EvHaus** [comment]
I'm also looking for this feature. In Yarn they call it `--all` (https://yarnpkg.com/cli/link#options-A%2C-all).

**Comment by jamesphilipps** [comment]
Any movement on this? I was very surprised to find this feature isn't already present. This makes pnpm link really fiddly in workspaces
